modules/social_features/social_group/modules/social_group_secret/src/EventSubscriber/SocialGroupSecretSubscriber.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\social_group_secret\EventSubscriber;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\EventSubscriber\CustomPageExceptionHtmlSubscriber;
+use Drupal\Core\EventSubscriber\DefaultExceptionHtmlSubscriber;
+use Drupal\Core\Path\PathMatcherInterface;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\group\Entity\GroupInterface;
+use Drupal\r4032login\EventSubscriber\R4032LoginSubscriber;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Class SocialGroupSecretSubscriber.
+ *
+ * @package Drupal\social_group_secret\EventSubscriber
+ */
+class SocialGroupSecretSubscriber extends R4032LoginSubscriber {
```

Comment:
I don't think we need to inherit the `R4032LoginSubscriber` here? The event system should be flexible enough to allow us to simply inherit from `HttpExceptionSubscriberBase` and also listen to the on403 event, returning a 404 instead.

Is there something specific we use from R4032Login that we need to use?

Reply:
@Kingdutch we should inherit it because we should have a response for the not secret group pages created inherited class.

Reply:
@OH17L I'm not sure I understand. Are you saying there can only be one handler of the on403 event? I thought multiple listeners would be possible? Then we can try and handle the exception if it's for a secret group and otherwise we pass it on to the next listener.

Reply:
If you're looking for an example for the future, this has been solved in 1b452b05e